plugin/trino-exchange/src/main/java/io/trino/plugin/exchange/ExchangeStorageWriter.java

```diff
@@ -0,0 +1,29 @@
+package io.trino.plugin.exchange;
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import io.airlift.slice.Slice;
+
+import java.io.IOException;
+
+public interface ExchangeStorageWriter
+        extends AutoCloseable
+{
+    void write(Slice slice)
```

Comment:
I think this should allow for asynchronous IO. I.e it should return Future which denotes if writer buffers are depleted and caller should pause feeding the Writer with more data.

Then based on the futures returned from writers you can derive the future which you return from `FilesystemExchangeSInk.isBlocked()` (currently you always return NOT_BLOCKED there).

This can be a followup but IMO worth putting on backlog.

Reply:
This is good point. Do we only block when all partition are blocked because of uploads in progress? Or the API should be more finer grained at a partition level?

Reply:
That is a good question. As we do not control for what partition there will be data upcoming then probably we should return BLOCKED from ES as soon as any Writer is blocked. 
We may consider parametriziong `ES.isBlocked()` with partitionId. But let's proof test simpler solution first.